src/searchbar.cpp

```diff
@@ -97,6 +44,52 @@ SearchBar::SearchBar(QWidget *parent) :
     connect(this, &QLineEdit::returnPressed, this, [=]() {
         m_returnPressed = true;
     });
+
+    // placing an icon inside QLineEdit
+    addAction(m_action, QLineEdit::LeadingPosition);
+    setSearchMode();
+    connect(m_action, &QAction::triggered,
+            this, &SearchBar::on_action_triggered);
+}
+
+
+void SearchBar::setSearchMode()
+{
+    clear();    // prepare for entering search request
+    setReadOnly(false);
+
+    m_action->setEnabled(false);    // ignore clicking on 'Search' icon
+
+    m_action->setIcon(QIcon(":/icons/search.svg"));
+
+    /* Enable blinking cursor in QLineEdit
+     *
+     * There is no method to do it: QLineEdit doens't have API for controlling
+     * showing of text cursor.
+     *
+     * This workaround was found in QLineEdit sources:
+     * d->control->setBlinkingCursorEnabled(true)
+     * is called only from two places:
+     *  - from focusInEvent() - but we have to ignore it: at the time the event arrived
+     *    it could be action toggling next;
+     *  - from event() if (e->type() == QEvent::Show) - we use it here:
+     */
+    QApplication::postEvent(this, new QShowEvent());
+}
+
+void SearchBar::setBookmarkMode()
+{
+    setText(m_title);   // turn off editing, show the title of current tab
+    setReadOnly(true);
+
+    m_action->setEnabled(true); // enable toggling bookmark for the current tab
+
+    auto kiwixApp = KiwixApp::instance();
+    if (kiwixApp->isCurrentArticleBookmarked()) {
+        m_action->setIcon(QIcon(":/icons/reading-list-active.svg"));
+    } else {
+        m_action->setIcon(QIcon(":/icons/reading-list.svg"));
+    }
 }
```

Comment:
I wonder if we could use two different action and set the action of the QLineEdit when we switch mode instead of reconfiguring the only one action.

Reply:
2 different actions would be indeed much simpler, but I tried to keep only one icon like it done in Apple Safari